server/src/main/java/com/genymobile/scrcpy/control/UhidManager.java

```diff
@@ -56,41 +55,35 @@ public UhidManager(DeviceMessageSender sender, String displayUniqueId) {
     }
 
     public void open(int id, int vendorId, int productId, String name, byte[] reportDesc) throws IOException {
-        try {
-            FileDescriptor fd = Os.open("/dev/uhid", OsConstants.O_RDWR, 0);
-            try {
-                // First UHID device added
-                boolean firstDevice = fds.isEmpty();
-
-                FileDescriptor old = fds.put(id, fd);
-                if (old != null) {
-                    Ln.w("Duplicate UHID id: " + id);
-                    close(old);
-                }
+        try (RandomAccessFile raf = new RandomAccessFile("/dev/uhid", "rw")) {
```

Comment:
`raf` will be closed by this `try`, so `raf.write` in `writeInput` will throw an error